### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by OyePriscilla**
## In this PR: I, 
- Created classes Classroom, Book and Rental.
- Created the has-many/belongs-to relationship between Classroom and Student.
- Created the many-to-many relationship between Person and Book using the intermediate class Rental.
- Setter and getter were created for the instance variables with the use of **attr_accessor**.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
+
+  def classroom=(classroom)
     @classroom = classroom
+    classroom.students.push(self)
   end
```
Comment (medaminedev66) [inline student.rb]: - [ ] You should add a condition that verifies if the "classroom.students" already have the student:
```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @OyePriscilla  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Highly organized/clean code.
- The repo looks professional.
- There are no linter errors.
- You have a descriptive PR.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (OyePriscilla) [reply student.rb]: Ok, noted with thanks.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @OyePriscilla,

✔️ Good documentation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**